src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdkAppEngineDevServer1.java

```diff
@@ -202,33 +202,32 @@ void checkAndWarnIgnored(@Nullable Object propertyToIgnore, String propertyName)
   }
 
   /**
-   * This method tries to guess the runtime based on the appengine-web.xml of all services that are
-   * expected to run.
+   * This method determines if the runtime based on the appengine-web.xml of all services that are
+   * expected to run can run without sandbox restrictions.
    *
    * @param services a list of app engine standard service directories
-   * @return {@code false} if only java7 modules are found or {@code true} is at least one java8
-   *     module is found (i.e. pure java8 or mixed java7/java8)
+   * @return {@code true} if there no need to enforce the sandbox restrictions.
    */
   @VisibleForTesting
-  boolean isJava8(List<File> services) throws AppEngineException {
-    boolean java8Detected = false;
+  boolean isWhiteListEnforced(List<File> services) throws AppEngineException {
+    boolean relaxSandbox = false;
     boolean java7Detected = false;
     for (File serviceDirectory : services) {
       Path appengineWebXml = serviceDirectory.toPath().resolve("WEB-INF/appengine-web.xml");
       try (InputStream is = Files.newInputStream(appengineWebXml)) {
-        if (AppEngineDescriptor.parse(is).isJava8()) {
-          java8Detected = true;
-        } else {
+        if (AppEngineDescriptor.parse(is).isWhiteListEnforced()) {
           java7Detected = true;
```

Comment:
I think also renaming `java7Detected` to `enforceSandbox` is better aligned with the contract of `isWhiteListEnforced()`, and less confusing overall.